.github/workflows/test_on_push.yml

```diff
@@ -198,20 +286,16 @@ jobs:
           python-version: 3.12
           cache: 'pip'
 
-      - name: Install nox
-        run: python -m pip install nox
+      - name: Install nox and tomlkit
+        run: python -m pip install nox tomlkit
 
       - name: Cache pybamm-requires nox environment for GNU/Linux
         uses: actions/cache@v4
         with:
           path: |
-            # Repository files
-            ${{ github.workspace }}/pybind11/
-            ${{ github.workspace }}/install_KLU_Sundials/
             # Headers and dynamic library files for SuiteSparse and SUNDIALS
-            ${{ env.HOME }}/.local/lib/
-            ${{ env.HOME }}/.local/include/
-          key: nox-${{ matrix.os }}-pybamm-requires-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/install_KLU_Sundials.py', '**/noxfile.py', '**/test_on_push.yml') }}
+            ${{ github.workspace }}/sundials_KLU_libs/
```

Comment:
Here, you cached `sundials_KLU_libs/` – I would suggest caching both folders and maintaining consistency for all Intel and macOS jobs.